ballerina/segment_writer.bal

```diff
@@ -74,11 +74,22 @@ isolated function writeSegment(map<json> seg, EdiSegSchema segMap, EdiContext co
             segLine += fd + repeatingText;
         } else {
             var fdata = seg.get(fieldSchema.tag);
+            if fieldSchema.length is EdiFieldLength {
+                EdiFieldLength fieldLength = <EdiFieldLength>fieldSchema.length;
+                if fieldLength.min > fdata.toString().length() {
+                    return error Error(string `Field length is less than the minimum length.
+                    Field: ${fieldSchema.tag}, Segment: ${segMap.tag}, Input value: ${fdata.toString()}, Minimum length: ${fieldLength.min}`);
+                }
+                if fieldLength.max < fdata.toString().length() && fieldLength.max != -1{
+                    return error Error(string `Field length is greater than the maximum length.
+                    Field: ${fieldSchema.tag}, Segment: ${segMap.tag}, Input value: ${fdata.toString()}, Maximum length: ${fieldLength.max}`);
+                }
+            }
```

Comment:
Cant we create separate function for this? Seems repeated

Reply:
The output error structure and values differ from segment writer to reader.

Reply:
Ack